### ISSUE TITLE
random_train_test_split doesn't split

### ISSUE THREAD

**Issue by jeydn** [lyst/lightfm#408]
Hi, 
I have a question regarding the random_train_test_split function. I'm trying to split the interactions into a train and test set, but it does not split it at all. This is my code: 

```
(train, test) = random_train_test_split(interactions, random_state=random_state, test_percentage=test_percentage)`

print('Shape: train %s, test %s.' % (train.shape, test.shape))

model = LightFM(loss='bpr', learning_rate=0.05) # learning_rate
model.fit(train, epochs=10) # item_features, epochs

train_auc = auc_score(model, train).mean()
test_auc = auc_score(model, test).mean()

print('AUC: train %.2f, test %.2f.' % (train_auc, test_auc))

```
In these are my console print statements: 
```
Num users: 610, num_items 9742.
<610x9742 sparse matrix of type '<class 'numpy.int32'>'
	with 100836 stored elements in COOrdinate format>
Shape: train (610, 9742), test (610, 9742).
AUC: train 0.90, test 0.86.
```

The interactions matrix has the shape of 610x9742. So, the same as train and test. 
Could maybe someone help me where I went wrong? 

Thanks a lot.

**Comment by maciejkula** [comment]
Though the matrix dimensions stay the same (as they must), the non-zero elements are split between the test and train matrices. Count the number of nonzero elements in each to confirm this.
